src/main/java/org/tikv/common/region/AbstractRegionStoreClient.java

```diff
@@ -147,6 +147,45 @@ public boolean onStoreUnreachable() {
       }
     }
 
+    Boolean result = trySwitchLeader();
+    if (result != null) {
+      return result;
+    }
+    if (conf.getEnableGrpcForward()) {
+      return tryGrpcForward();
+    }
+    return false;
+  }
+
+  protected Kvrpcpb.Context makeContext(TiStoreType storeType) {
+    return region.getReplicaContext(java.util.Collections.emptySet(), storeType);
+  }
+
+  protected Kvrpcpb.Context makeContext(Set<Long> resolvedLocks, TiStoreType storeType) {
+    return region.getReplicaContext(resolvedLocks, storeType);
+  }
+
+  private void updateClientStub() {
+    String addressStr = store.getStore().getAddress();
+    long deadline = timeout;
+    if (store.getProxyStore() != null) {
+      addressStr = store.getProxyStore().getAddress();
+      deadline = conf.getForwardTimeout();
+    }
+    ManagedChannel channel =
+        channelFactory.getChannel(addressStr, regionManager.getPDClient().getHostMapping());
+    blockingStub =
+        TikvGrpc.newBlockingStub(channel).withDeadlineAfter(deadline, TimeUnit.MILLISECONDS);
+    asyncStub = TikvGrpc.newFutureStub(channel).withDeadlineAfter(deadline, TimeUnit.MILLISECONDS);
+    if (store.getProxyStore() != null) {
+      Metadata header = new Metadata();
+      header.put(TiConfiguration.FORWARD_META_DATA_KEY, store.getStore().getAddress());
+      blockingStub = MetadataUtils.attachHeaders(blockingStub, header);
+      asyncStub = MetadataUtils.attachHeaders(asyncStub, header);
+    }
+  }
+
+  private Boolean trySwitchLeader() {
     Histogram.Timer switchLeaderDurationTimer = SWITCH_LEADER_DURATION.startTimer();
```

Comment:
how about renaming it to `find_leader_store_duration`?

Reply:
done